cli/injector/main.go

```diff
@@ -329,6 +339,50 @@ func injectAndWait(cmd *cobra.Command, args []string) {
 		}
 	}
 
+	if len(pulsingInjectors) > 0 {
+		isInjected := true
+		lastOperationTime := time.Now()
+
+		for { // This loop will inject, wait, clean, wait until a signal is received
+			// Inject all the pulsing disruptions
+			if time.Since(lastOperationTime) >= pulseDormantDuration && !isInjected {
```

Comment:
I think (either at the end or depending of this function) we can add a small sleep, as we should be able to calculate approximately the next time we need to either re-activate or re-dormant. Currently, we will just run these `if` checks extremely quickly. I think, to avoid an issue with catching the exit signal, it may be possible to put a sleep in the select conditions?

Something like:
```
select {
case <- signals:
case sleepTimerRanOut:
}
```

Reply:
Wouldn't it prevent the signal to be caught early on? Is it ok to wait for the sleep to finish to be able to remove the disruption?

Reply:
I don't think so? If the sleep is set up as a channel, then we're just blocking until we get either the exit signal or the wake up signal. Am I wrong? This way we're not running code for no reason